packages/amplify-graphql-api-construct/src/internal/asset-manager.ts

```diff
@@ -11,9 +12,13 @@ const RESOLVER_PREFIX = 'resolvers';
  * with the path method which is used in CDK.
  */
 export class AssetManager {
```

Comment:
I recognize that this is not an architecture that you're changing as part of this PR, but I don't understand the relationship between the construct's AssetManager and the AssetManager in the transformer core--despite their shared name, they have different interfaces (and presumably, responsibilities). Can you expand the doc comment above with some extra detail:
- What is the responsibility of each of the AssetManager classes?
- What is the relationship, if any, between the two?
- I don't see that the construct AM conforms to the transformer's AssetManagerProvider interface, so I'm unsure how the construct AM logic is actually invoked.

My hope is to come away with a high-level understanding of how the core asset generation logic uses the construct's AM

Reply:
I'll expand more later but the basic gist is that the AssetManager in the construct is utilized by the AssetManager in the transformer. The transformer AssetManager knows that it is supposed to store assets, but it doesn't know the details of how this is done. The transformer AssetManager relies on the construct AssetManager to implement the details.

The `createAsset` function in the transformer AssetManager:
https://github.com/aws-amplify/amplify-category-api/blob/a5a233cfc1aa0859a9adf74f319b53c9479b38d4/packages/amplify-graphql-transformer-core/src/transformer-context/asset-manager.ts#L7-L9

is calling the provide function:
https://github.com/aws-amplify/amplify-category-api/blob/b044a26366653e4c505b040a7809e3bd56471594/packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts#L193-L194

Which is then calling the `addAsset` function from the construct AssetManager:
https://github.com/aws-amplify/amplify-category-api/blob/b044a26366653e4c505b040a7809e3bd56471594/packages/amplify-graphql-api-construct/src/internal/asset-manager.ts#L22-L32